doc/source/io.rst

```diff
@@ -3941,56 +3961,165 @@ destination DataFrame as well as a preferred column order as follows:
 
 .. code-block:: python
 
-   df = pandas.DataFrame({'string_col_name' : ['hello'],
+   df = pd.DataFrame({'string_col_name' : ['hello'],
          'integer_col_name' : [1],
          'boolean_col_name' : [True]})
-   df.to_gbq('my_dataset.my_table', project_id = projectid)
+   pd.to_gbq(df, 'my_dataset.my_table', projectid, if_exists='fail')
+
+.. note::
 
-The BigQuery SQL query language has some oddities, see `here <https://developers.google.com/bigquery/query-reference>`__
+    If the ``if_exists`` argument is set to ``append``, the destination dataframe will
+    be written to the table using the defined table schema and column types. The
+    dataframe must match the destination table in column order, structure, and
+    data types.
+    If the ``if_exists`` argument is set to ``replace``, and the existing table has a
+    different schema, a delay of 2 minutes will be forced to ensure that the schema
+    has propagated in the Google environment. See
+    `Google BigQuery issue 191 <https://code.google.com/p/google-bigquery/issues/detail?id=191>`__
 
-While BigQuery uses SQL-like syntax, it has some important differences
-from traditional databases both in functionality, API limitations (size and
-quantity of queries or uploads), and how Google charges for use of the service.
-You should refer to Google documentation often as the service seems to
-be changing and evolving. BiqQuery is best for analyzing large sets of
-data quickly, but it is not a direct replacement for a transactional database.
+Writing large DataFrames can result in errors due to size limitations being exceeded.
+This can be avoided by setting the ``chunksize`` argument when calling :func:`~pandas.io.to_gbq`.
+For example, the following writes ``df`` to a BigQuery table in batches of 10000 rows at a time:
 
-You can access the management console to determine project id's by:
-<https://code.google.com/apis/console/b/0/?noredirect>
+.. code-block:: python
+
+    pd.to_gbq(df, 'my_dataset.my_table', projectid, chunksize=10000)
 
-As of 0.15.2, the gbq module has a function ``generate_bq_schema`` which
-will produce the dictionary representation of the schema.
+You can also see the progess of your post via the ``verbose`` flag which defaults to ``True``.
+For example:
 
 .. code-block:: python
 
-   df = pandas.DataFrame({'A': [1.0]})
-   gbq.generate_bq_schema(df, default_type='STRING')
+    pd.to_gbq(df, 'my_dataset.my_table', projectid, chunksize=10000, verbose=True)
 
-.. warning::
+.. note::
 
-   To use this module, you will need a valid BigQuery account. See
-   <https://cloud.google.com/products/big-query> for details on the
-   service.
+    The http response code of Google BigQuery can be successful (200) even if the append failed.
+    For this reason, if there is a failure to append to the table, the complete error response
+    from BigQuery is returned which can be quite long given it provides a status for each row.
```

Comment:
why is this? u should then also check the response and raise an exception of this happened
silent failing is bad